src/components/uploader/layers/canvas.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useRef, useState, PointerEvent } from 'react';
+import { v4 as uuidV4 } from 'uuid';
+
+import CanvasHelper from './canvas.helper';
+import { UploaderEnum, useUploader } from '../../../reducers';
+
+import styles from '../uploader.module.css';
+import { Point } from '../../../types';
+
+export const UploaderLayerCanvas = () => {
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+  const [ctx, setCtx] = useState<CanvasRenderingContext2D>();
+  const [startPoint, setStartPoint] = useState(CanvasHelper.createOrigin());
+  const [endPoint, setEndPoint] = useState(CanvasHelper.createOrigin());
+  const [isDragging, setIsDragging] = useState(false);
+  const { uploader, dispatch } = useUploader();
+  useEffect(() => {
+    const context = canvasRef.current?.getContext(
+      '2d'
+    ) as CanvasRenderingContext2D;
+    setCtx(context);
+  }, [canvasRef]);
+  useEffect(() => {
+    if (!ctx) {
+      return;
+    }
+    ctx.clearRect(
+      0,
+      0,
+      uploader.stageWidth as number,
+      uploader.stageHeight as number
+    );
+    uploader.coordinates.forEach((coordinate) => {
+      drawShape(CanvasHelper.makeVerticesFromCoordinate(coordinate));
+    });
+  }, [uploader.coordinates]);
```

Comment:
I found `react-hooks/exhaustive-deps` warnings at this line. Is it intentional?

Reply:
Yes, it's intentional as I only want to run `useEffect` when `coordinates` are updated. But I think there's a better implementation to fix the warning as well. Let me fix this.

Reply:
Fixed in [ad1b08c](https://github.com/chatbotgang/image-map-editor/pull/22/commits/ad1b08c6ff1a38b72e5421a7ba1a82603f42813f)